<a href="https://colab.research.google.com/github/camilaccosta/Exerc-cios-m-dulo-1-/blob/main/L_Pandas2_Q4e5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Questão 4**

O arquivo googleplaystore.csv é um dataset que contém diversas informações sobre aplicativos disponíveis para download na Google Play Store. Logo após, verifique se existem dados duplicados e/ou ausentes nesse dataset.

Caso existam dados duplicados, remova-os. Caso existam dados ausentes, você pode removê-los ou pensar em alguma maneira de preenchê-los.



In [23]:
import pandas as pd 


In [3]:
google_play = pd.read_csv("googleplaystore.csv")

**Verifique se existem dados duplicados e/ou ausentes nesse dataset.**

**Caso existam dados duplicados, remova-os.**

In [18]:
google_play.drop_duplicates("App", inplace=True)

**Caso existam dados ausentes, você pode removê-los ou pensar em alguma maneira de preenchê-los.**

In [21]:
google_play.dropna(inplace=True)

In [22]:
google_play.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

**QUESTÃO 5**

O arquivo houses_to_rent.csv consiste em um dataset que contém dados sobre casas para alugar no Brasil. Carregue-o em um objeto DataFrame e verifique os dados contidos nas colunas desse DF (você pode utilizar os métodos head, tail e info). Logo após, verifique se existem dados duplicados e/ou ausentes nesse dataset.

Caso existam dados duplicados, remova-os. Caso existam dados ausentes, você pode removê-los ou pensar em alguma maneira de preenchê-los.

In [24]:
# Realizando a leitura do arquivo 
houses_to_rent=pd.read_csv("houses_to_rent.csv" ) 

In [44]:
# Fazendo um filtro a fim de verificar se existem linhas duplicadas.

houses_to_rent[houses_to_rent.duplicated()== True]


city  area  ...  fire insurance (R$)  total (R$)
244      Porto Alegre    47  ...                   33        2633
920    Belo Horizonte    40  ...                   13         983
1060        São Paulo   136  ...                   54        6054
1064        São Paulo    20  ...                   23        2555
1108         Campinas   184  ...                   49        3509
...               ...   ...  ...                  ...         ...
10560    Porto Alegre    40  ...                   18        2586
10569  Belo Horizonte    20  ...                   15        1115
10623    Porto Alegre    47  ...                   33        2633
10640       São Paulo    27  ...                   45        4951
10641  Belo Horizonte    58  ...                   14        1512

[358 rows x 13 columns]

*No output anterior foi possível verificar que 358 linhas eram duplicadas.* 

**Caso existam dados duplicados, remova-os.**

In [52]:
#Removendo as linhas duplicadas. O parâmetro keep = "first", faz com que a primeira linha duplicada seja mantida e as demais apagadas. 
houses_to_rent.drop_duplicates(keep="first")

city  area  ...  fire insurance (R$)  total (R$)
0           São Paulo    70  ...                   42        5618
1           São Paulo   320  ...                   63        7973
2        Porto Alegre    80  ...                   41        3841
3        Porto Alegre    51  ...                   17        1421
4           São Paulo    25  ...                   11         836
...               ...   ...  ...                  ...         ...
10687    Porto Alegre    63  ...                   22        1926
10688       São Paulo   285  ...                  191       19260
10689  Rio de Janeiro    70  ...                   78        7390
10690  Rio de Janeiro   120  ...                  155       14020
10691       São Paulo    80  ...                   22        1587

[10334 rows x 13 columns]

In [50]:
#Verificando se tem algum valor NaN no Data Frame.
houses_to_rent.isnull().values.any()

False